### PR TITLE
[TT-7820] fix: Fix aggregate graph pump sharding and errors

### PR DESCRIPTION
**Pull request by kofoworola**
## Description
This PR fixes the issue with the aggregate pump sharding and the error handling specified in the ticket
## Related Issue

[TT-7820](https://tyktech.atlassian.net/browse/TT-7820)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

[TT-7820]: https://tyktech.atlassian.net/browse/TT-7820?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by tbuchaillot** [comment]
@kofoworola tests are failing. Can you check that before the review?

**Comment by kofoworola** [comment]
> 1 Code Smell

@tbuchaillot it's been fixed

**Comment by kofoworola** [comment]
> @kofoworola tests are failing. Can you check that before the review?

@tbuchaillot yeah it's all good now.

File: analytics/graph_record.go [hunk 1]
```diff
@@ -30,29 +31,29 @@ type GraphRecord struct {
...
 
-func (a *AnalyticsRecord) ToGraphRecord() (GraphRecord, error) {
-	record := GraphRecord{
-		AnalyticsRecord: *a,
-		RootFields:      make([]string, 0),
-		Types:           make(map[string][]string),
-	}
-	if a.ResponseCode >= 400 {
-		record.HasErrors = true
+func parseRequest(encodedRequest, encodedSchema string, record *GraphRecord) {
```
Comment (tbuchaillot) [inline analytics/graph_record.go:34]: @kofoworola please let's refactor this to be part of `GraphRecord` and add some comments 🙏

File: analytics/graph_record.go [hunk 2]
```diff
@@ -84,54 +86,76 @@ func (a *AnalyticsRecord) ToGraphRecord() (GraphRecord, error) {
...
-		if err != nil {
-			return record, nil
-		}
-		resp, err := http.ReadResponse(bufio.NewReader(bytes.NewReader(responseDecoded)), nil)
-		if err != nil {
-			log.WithError(err).Error("error reading raw response")
-			return record, err
-		}
-		defer resp.Body.Close()
+func parseResponse(encodedResponse string, record *GraphRecord) {
```
Comment (tbuchaillot) [inline analytics/graph_record.go:103]: same as `parseRequest`

**Review by tbuchaillot — CHANGES_REQUESTED** [review]
@kofoworola small comment - everything else looks 👌

File: analytics/graph_record.go (hunk 1 above)
Comment (kofoworola) [reply analytics/graph_record.go]: Thanks for pointing this, addressed

**Review by tbuchaillot — APPROVED** [review]
lgtm 🚀

**Review by pvormste — APPROVED** [review]

**Review by buraksezer — APPROVED** [review]

**Pull request merged**